plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js

```diff
@@ -504,64 +499,52 @@ jQuery( function ( $ ) {
 		}
 	}
 
-	$( 'select.wc-attribute-search' ).on( 'select2:select', function ( e ) {
-		if ( e.params && e.params.data && e.params.data.id ) {
-			add_attribute( this, e.params.data.id );
-			if ( ! selectedAttributes.includes( e.params.data.id ) ) {
-				selectedAttributes.push( e.params.data.id );
-				$( 'select.wc-attribute-search' ).data(
-					'disabled-items',
-					selectedAttributes
-				);
+	function add_if_not_exists( arr, item ) {
+		return arr.includes( item ) ? attr : [ ...arr, item ];
+	}
+
+	function disable_in_attribute_search( selectedAttributes ) {
+		$( 'select.wc-attribute-search' ).data( 'disabled-items', selectedAttributes );
+	}
+
+	function remove_empty_attribute_if_only() {
```

Comment:
I think I understood that `if_only` is because of the `if ( $attributes.length === 1 )` part. But that's still confusing to me. However I'm not sure about a better name. Maybe `remove_empty_attribute_if_only_one`?

Reply:
Good idea. I'll go with `remove_blank_custom_attribute_if_no_other_attributes`.